cdn-server/cdn/src/index.ts

```diff
@@ -88,13 +89,57 @@ const persistedOperation = (storage: BlobStorage) => {
   };
 };
 
+const routerConfig = (storage: BlobStorage) => {
+  return async (c: Context) => {
+    const organizationId = c.req.param('organization_id');
+    const federatedGraphId = c.req.param('federated_graph_id');
+    // Check authentication
+    if (
+      organizationId !== c.get('authenticatedOrganizationId') ||
+      federatedGraphId !== c.get('authenticatedFederatedGraphId')
+    ) {
+      return c.text('Forbidden', 403);
```

Comment:
This is semantically a 400 (Bad request). Auth is handled by jwt middleware.

Reply:
Done